RUFAS/biophysical/manure/separators/separator.py

```diff
@@ -0,0 +1,277 @@
+
+
+class Separator(Processor):
+    """
+    A manure processor that separates a portion of solids from manure.
+
+    Parameters
+    ----------
+    name : str = ""
+        The name of the separator.
+    percent_dry_solids : float = 0.0
+        The dry matter content (percent DM) of separated manure solids.
+    ammoniacal_nitrogen_efficiency : float = 0.0
+        The efficiency of the separator in removing ammoniacal nitrogen from the manure.
+    nitrogen_efficiency : float = 0.0
+        The efficiency of the separator in removing nitrogen from the manure.
+    phosphorus_efficiency : float = 0.0
+        The efficiency of the separator in removing phosphorus from the manure.
+    potassium_efficiency : float = 0.0
+        The efficiency of the separator in removing potassium from the manure.
+    ash_efficiency : float = 0.0
+        The efficiency of the separator in removing ash from the manure.
+    volatile_solids_efficiency : float = 0.0
+        The efficiency of the separator in removing volatile solids from the manure.
+    total_solids_efficiency : float = 0.0
+        The efficiency of the separator in removing total solids from the manure.
+
+    Attributes
+    ----------
+    is_housing_emissions_calculator : bool = False
+        A flag to indicate if the processor is a housing emissions calculator. Will always be false for a separator.
+    name : str
+        The name of the separator.
+    prefix : str
+        The prefix to be used for naming output variables.
+    percent_dry_solids : float
+        The dry matter content (percent DM) of separated manure solids.
+    ammoniacal_nitrogen_efficiency : float
+        The efficiency of the separator in removing ammoniacal nitrogen from the manure.
+    nitrogen_efficiency : float
+        The efficiency of the separator in removing nitrogen from the manure.
+    phosphorus_efficiency : float
+        The efficiency of the separator in removing phosphorus from the manure.
+    potassium_efficiency : float
+        The efficiency of the separator in removing potassium from the manure.
+    ash_efficiency : float
+        The efficiency of the separator in removing ash from the manure.
+    volatile_solids_efficiency : float
+        The efficiency of the separator in removing volatile solids from the manure.
+    total_solids_efficiency : float
+        The efficiency of the separator in removing total solids from the manure.
+    om : OutputManager
+        The output manager instance used to store and manage output data.
+
+    """
+
+    def __init__(
+        self,
+        name: str = "",
+        percent_dry_solids: float = 0.0,
+        ammoniacal_nitrogen_efficiency: float = 0.0,
+        nitrogen_efficiency: float = 0.0,
+        phosphorus_efficiency: float = 0.0,
+        potassium_efficiency: float = 0.0,
+        ash_efficiency: float = 0.0,
+        volatile_solids_efficiency: float = 0.0,
+        total_solids_efficiency: float = 0.0,
```

Comment:
```suggestion
        percent_dry_solids: float,
        ammoniacal_nitrogen_efficiency: float,
        nitrogen_efficiency: float,
        phosphorus_efficiency: float,
        potassium_efficiency: float,
        ash_efficiency: float,
        volatile_solids_efficiency: float,
        total_solids_efficiency: float,
```
I suggest not having any defaults for these values. If there are any defaults we want applied, I think they should be defined in the metadata properties and only in the metadata properties.

Alternatively, if we want certain defaults or hardcoded values for certain types of processors, then I think it would be best to put them in those the child classes.